packages/core/security/core-security-browser/src/api_provider.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { CoreAuthenticationService } from './authc';
+
+/**
+ * The internal contract exposed by the security provider for Core to
+ * consume and re-expose via its security service.
+ *
+ * @public
+ */
+export interface CoreInternalSecurityContract {
+  authc: InternalAuthenticationServiceContract;
+}
```

Comment:
This is the interface for the shape of the API as registered from the security plugin to Core. ATM it just reflects the public interface of Core's security service's start contract, but this will change in later phase.

FWIW, the name is absolutely terrible (because we all know that naming is hard). Please tell me if anyone got a better idea.

Reply:
Yeah, as I said the name is terrible. You're totally right that we're keeping the `internal` qualifier for internal types only. I will rename it.